README.adoc

```diff
@@ -12,129 +12,132 @@ image:https://img.shields.io/github/license/openshift/odo?style=for-the-badge[Li
+* Completely client based. No additional server other than Kubernetes or OpenShift is required for deployment.
 * Official support for Node.js and Java components.
-* Partial compatibility with languages and frameworks such as Ruby, Perl, PHP, and Python. 
 * Detects changes to local code and deploys it to the cluster automatically, giving instant feedback to validate changes in real time.
 * Lists all the available components and services from the cluster.
 
-[id="odo-supported-languages-and-images"]
-=== Officially supported languages and corresponding container images
+[[core-concepts]]
+== Core concepts
 
-.Supported container images
-[cols=",,,",options="header",]
-|===
-|Language |Container Image |Supported Package Manager |Supported Platform
-|*Node.js*
-|https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/nodejs-10-rhel7[rhscl/nodejs-10-rhel7]
-|NPM
-|amd64, s390x, ppc64le
-
-|
-|https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/nodejs-12-rhel7[rhscl/nodejs-12-rhel7]
-|NPM
-|amd64, s390x, ppc64le
-
-|
-|https://access.redhat.com/articles/3376841[rhoar-nodejs/nodejs-10]
-|NPM
-|amd64
-
-|
-|https://github.com/sclorg/s2i-nodejs-container[centos/nodejs-10-centos7]
-|NPM
-|amd64
-
-|
-|https://github.com/sclorg/s2i-nodejs-container[centos/nodejs-12-centos7]
-|NPM
-|amd64
-
-|*Java*
-|https://access.redhat.com/containers/#/registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift[redhat-openjdk-18/openjdk18-openshift]
-|Maven, Gradle
-|amd64, s390x, ppc64le
-
-|
-|https://access.redhat.com/containers/#/registry.access.redhat.com/openjdk/openjdk-11-rhel8[openjdk/openjdk-11-rhel8]
-|Maven, Gradle
-|amd64, s390x, ppc64le
-
-|
-|https://access.redhat.com/containers/#/registry.access.redhat.com/openjdk/openjdk-11-rhel7[openjdk/openjdk-11-rhel7]
-|Maven, Gradle
-|amd64, s390x, ppc64le
+Project::
+A project is your source code, tests, and libraries organized in a separate single unit.
+Application::
+An application is a program designed for end users. An application consists of multiple microservices or components that work individually to build the entire application.
+Examples of applications: a video game, a media player, a web browser.
+Component::
+A component is a set of Kubernetes resources which host code or data. Each component can be run and deployed separately.
+Examples of components: Node.js, Perl, PHP, Python, Ruby.
+Service::
+A service is software that your component links to or depends on.
+Examples of services: MariaDB, Jenkins, MySQL.
+Devfile::
+A portable file responsible for your entire reproducable development environment.
+
+[id="odo-supported-devfiles"]
+=== Official Devfiles
+
+Devfiles describe your development environment link. link:https://odo.dev/docs/deploying-a-devfile-using-odo/[Click here for more information on Devfile.]
+
+All official example Devfiles are hosted on the link:https://github.com/odo-devfiles/registry[registry].
```

Comment:
I would rather not mention this url. As this is not an official Devfile registry. it is just the one that we are currently using until there is an official one that we share with Che.